document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/ArrayIdentifierExpression.java

```diff
@@ -14,28 +14,132 @@
 @EqualsAndHashCode(callSuper = true)
 public class ArrayIdentifierExpression extends IdentifierExpression {
 
-  private final ArrayType arrayType;
+  private final FlatCollectionDataType arrayElementType;
 
   public ArrayIdentifierExpression(String name) {
     this(name, null);
   }
 
-  public ArrayIdentifierExpression(String name, ArrayType arrayType) {
-    super(name);
-    this.arrayType = arrayType;
+  // Package-private: used internally by factory methods
+  ArrayIdentifierExpression(String name, FlatCollectionDataType arrayElementType) {
+    super(name, null);
+    this.arrayElementType = arrayElementType;
   }
 
   public static ArrayIdentifierExpression of(String name) {
     return new ArrayIdentifierExpression(name);
   }
 
-  public static ArrayIdentifierExpression of(String name, ArrayType arrayType) {
-    return new ArrayIdentifierExpression(name, arrayType);
+  // Package-private: used internally by factory methods
+  static ArrayIdentifierExpression of(String name, FlatCollectionDataType arrayElementType) {
+    return new ArrayIdentifierExpression(name, arrayElementType);
   }
 
-  /** Returns the array type if specified, empty otherwise */
-  public Optional<ArrayType> getArrayType() {
-    return Optional.ofNullable(arrayType);
+  public static ArrayIdentifierExpression ofStrings(final String name) {
+    return of(name, PostgresDataType.TEXT);
+  }
+
+  public static ArrayIdentifierExpression ofInts(final String name) {
+    return of(name, PostgresDataType.INTEGER);
+  }
+
+  public static ArrayIdentifierExpression ofLongs(final String name) {
+    return of(name, PostgresDataType.BIGINT);
+  }
+
+  public static ArrayIdentifierExpression ofShorts(final String name) {
+    return of(name, PostgresDataType.SMALLINT);
+  }
+
+  public static ArrayIdentifierExpression ofFloats(final String name) {
+    return of(name, PostgresDataType.FLOAT);
+  }
+
+  public static ArrayIdentifierExpression ofDoubles(final String name) {
+    return of(name, PostgresDataType.DOUBLE);
+  }
+
+  public static ArrayIdentifierExpression ofDecimals(final String name) {
+    return of(name, PostgresDataType.NUMERIC);
+  }
+
+  public static ArrayIdentifierExpression ofBooleans(final String name) {
+    return of(name, PostgresDataType.BOOLEAN);
+  }
+
+  public static ArrayIdentifierExpression ofTimestamps(final String name) {
+    return of(name, PostgresDataType.TIMESTAMP);
+  }
+
+  public static ArrayIdentifierExpression ofTimestampsTz(final String name) {
+    return of(name, PostgresDataType.TIMESTAMPTZ);
+  }
+
+  public static ArrayIdentifierExpression ofDates(final String name) {
+    return of(name, PostgresDataType.DATE);
+  }
+
+  public static ArrayIdentifierExpression ofUuids(final String name) {
```

Comment:
Do we need to create a public method for every Postgres type here? I'd start with minimal types what is required (and common across different database types) and live with it and start adding more on a need basis.